CardsCashCasino/Manager/TexasHoldEmManager.cs

```diff
@@ -0,0 +1,172 @@
+﻿/*
+ *  Module Name: TexasHoldEmManager.cs
+ *  Purpose: Manages the game of Texas Hold Em.
+ *  Inputs: None
+ *  Outputs: None
+ *  Additional code sources: None
+ *  Developers: Mo Morgan
+ *  Date: 11/32024
+ *  Last Modified: 11/3/2024
+ *  Preconditions: None
+ *  Postconditions: None
+ *  Error/Exception conditions: None
+ *  Side effects: None
+ *  Invariants: None
+ *  Known Faults: None encountered
+ */
+
+using System;
+using System.Collections.Generic;
+using CardsCashCasino.Data;
+
+namespace CardsCashCasino.Manager
+{
+    /// <summary>
+    /// The possible actions that a player can take in Texas Hold Em.
+    /// </summary>
+    public enum PokerAction
+    {
+        FOLD,
+        CHECK,
+        CALL,
+        RAISE,
+        ALL_IN
+    }
+
+    /// <summary>
+    /// The possible roles that a player can have in Texas Hold Em.
+    /// The Small Blind is always left of the Dealer.
+    /// </summary>
+    public enum PlayerRole
+    {
+        DEALER,
+        SMALL_BLIND,
+        BIG_BLIND,
+    }
+    
+    public class TexasHoldEmManager
+    {
+        /// <summary>
+        /// The cardManager for the game.
+        /// </summary>
+        private CardManager _cardManager;
```

Comment:
Let's not initialize card manager per game manager. Instead, we have a SINGLE card manager that interacts with each game manager. If we initialize per class, it makes the concept of a card manager redundant and makes communication between games hard. Inter-game communication for cards is likely not super relevant, but is for ChipManager which should also have this treatment.

Reply:
If we don't initialize a CardManager within this class, we can't access it's `Cards` object to draw cards from. If we're going to only have one CardManager, we'll need to decouple `Cards` and `DrawCard()` from it, which would defeat the purpose.